### DISCUSSION TITLE
Remove the use of Moq or no update

### DISCUSSION THREAD

**Discussion by cfonseca23** [OrchardCMS/OrchardCore#14136] [General]
**Is your feature request related to a problem? Please describe.**
Moq 4.20 and later collects telemetry data that we do not wish to participate in:

https://www.bleepingcomputer.com/news/security/popular-open-source-project-moq-criticized-for-quietly-collecting-data/

https://yro.slashdot.org/story/23/08/09/2058205/popular-open-source-project-moq-criticized-for-quietly-collecting-data

https://github.com/moq/moq/commit/6057dd207f597db2b8f9acc74f1fec9c8c0a2655
https://github.com/devlooped/SponsorLink/commit/a199a4d88afccbeaba5bfe6078e037f56d60ccff

**Describe the solution you'd like**
Remove Moq

-  thanks to "@lukebakken"

**Comment by hishamco** [comment]
This is a decision that the team should make. Personally, I never mind using Moq and avoid updating the affected version

**Comment by Tony20221** [comment]
> **Is your feature request related to a problem? Please describe.** Moq 4.20 and later collects telemetry data that we do not wish to participate in:
> 
> https://www.bleepingcomputer.com/news/security/popular-open-source-project-moq-criticized-for-quietly-collecting-data/
> 
> https://yro.slashdot.org/story/23/08/09/2058205/popular-open-source-project-moq-criticized-for-quietly-collecting-data
> 
> [moq/moq@6057dd2](https://github.com/moq/moq/commit/6057dd207f597db2b8f9acc74f1fec9c8c0a2655) [devlooped/SponsorLink@a199a4d](https://github.com/devlooped/SponsorLink/commit/a199a4d88afccbeaba5bfe6078e037f56d60ccff)
> 
> **Describe the solution you'd like** Remove Moq
> 
> * thanks to "@lukebakken"

The Moq issue is very recent. The second people learned about the telemetry thing, a ton of people are into a panic, 'remove Moq', 'Switch to NSubtitute' blah blah modes!

Let the issue's dust settle first and let us know what Moq's plans are. Moq's author lately has been explaining stuff and AFAIK, he reverted the change.

**Comment by Piedone** [comment]
Orchard uses Moq v4.18.4, so not the version affected (>4.20.0). However, personally, I wouldn't like to see sponsorship requests in the build output, nor as a company owner find it suitable to ask for sponsorship connected to each developer's e-mail.
